FrontEnd/src/components/profileList/ProfileListPage.jsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { Radio } from 'antd';
+import useSWR from 'swr';
+
+import ErrorPage404 from '../errorPages/ErrorPage404';
+import Loader from '../loader/Loader';
+import ProfileList from './ProfileList';
+
+import css from './ProfileListPage.module.css';
+
+export default function ProfileListPage({ isAuthorized }) {
+  const { filter } = useParams();
+
+  const [filterSaved, setFilterSaved] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const [profileFilter, setProfileFilter] = useState('');
+
+  useEffect(() => {
+    switch (filter) {
+      case 'companies':
+        setProfileFilter('is_registered=True');
+        break;
+      case 'startups':
+        setProfileFilter('is_startup=True');
+        break;
+      case 'producers':
+        setProfileFilter('activities__name=Виробник');
+        break;
+      case 'importers':
+        setProfileFilter('activities__name=Імпортер');
+        break;
+      case 'retailers':
+        setProfileFilter('activities__name=Роздрібна мережа');
+        break;
+      case 'horeca':
+        setProfileFilter('activities__name=HORECA');
```

Comment:
In order to remove the switch case you may create map for it.
 `
 const FILTER_MAP = {
    companies: 'is_registered=True',
    startups: 'is_startup=True',
    ...
}
 `
and use it in one line 
`setProfileFilter(FILTER_MAP[filter])`

Reply:
Wow, thanks. I really wanted to get rid of that switch-case.